staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/versioned/conversion.go

```diff
@@ -28,6 +28,17 @@ type Convertor struct {
 	Scheme *runtime.Scheme
 }
 
+// Convert converts the in object to the out object and returns an error if the
+// conversion fails.
+func (c Convertor) Convert(in runtime.Object, out runtime.Object) error {
+	// for custom resources, because ConvertToGVK reuses the passed in object as
+	// the output, so they can skip the conversion here.
+	if in == out {
```

Comment:
This is strictly an optimization and not a correctness thing, right? Can you add this detail to the comment?

Reply:
It's also for correctness. Currently the conversion stack will set the object to empty if we send the same runtime.Object as `in` and `out` to `Convert(in, out, context)`.

Will add that into the comment.

Reply:
Done.